### PR TITLE
Systemd env

### PR DESCRIPTION
**Pull request by nicolasbock**
Please help potential reviewers to understand this pull request and speed
up the process by writing a meaningful pull request message.

Answering the following questions can help, but is optional.

**Why is this change necessary?**

**How does it address the issue?**

**Is there additional information worth sharing like links to a Trello
card, bug references, testing advice or dependencies to other pull
requests?**

### CODE REVIEW

**Comment by Itxaka** [comment]
I did something similar to allow doing override files for systemd services with extra configs in https://github.com/crowbar/crowbar-core/pull/1418

My issue was that I didnt want to duplicate the code already existing, because if Im not mistaken, most of this code is duplicated on systemd_service_restart.rb

So if you see on my PR @vuntz suggested expanding systemd_service_restart.rb with the new action (in this case `environment_file`?) and creating a middle layer to maintain compatibility with the cookbooks calling `systemd_service_restart`

Although Im not sure if there is time constraints for this feature?

**Comment by nicolasbock** [comment]
Thanks @Itxaka ! I did basically copy the `restart` code :wink: The only time constraint in a way is that https://github.com/crowbar/crowbar-openstack/pull/614 depends on this one and is blocked that way until your PR gets merged. But on the other hand, your approach sounds much cleaner. Let's get https://github.com/crowbar/crowbar-core/pull/1418 to move forward then!

**Comment by Itxaka** [comment]
@nicolasbock take into account that #1418 is closed because it was not needed anymore, so I did not develope further and it will need some work, while this is already done.... so if it needs to get in to unblock https://github.com/crowbar/crowbar-openstack/pull/614 I will say lets move with this and create a card/bug/whatever for converging both resources into a new one.

**Comment by nicolasbock** [comment]
Ok, that sounds good. I have tested this resource with the midonet one on top and it works as advertised.

**Comment by Itxaka** [comment]
@nicolasbock looks like github malfunctioned and no tests are being run :(

**Comment by nicolasbock** [comment]
I have restarted the Jenkins build. Hopefully it will work now.

**Comment by nicolasbock** [comment]
Rebased

**Comment by nicolasbock** [comment]
There are some HA failures, but I don't see how those are related or caused by this PR>

**Comment by nicolasbock** [comment]
Rebased.

**Comment by vuntz** [comment]
If you feel brave, as suggested by @Itxaka, I think it makes sense to merge the LWRPs, and provide a compatibility layer for the old existing one.

**Comment by nicolasbock** [comment]
While I am working on refactoring the `systemd` providers, could we merge this one to unblock https://github.com/crowbar/crowbar-openstack/pull/614 ?

**Comment by nicolasbock** [comment]
@Itxaka and @AbelNavarro Could you re-review this PR? I would prefer if we merge it this way to unblock https://github.com/crowbar/crowbar-openstack/pull/614 and I will work on refactoring the `systemd` providers. Thanks!

File: chef/cookbooks/utils/providers/systemd_environment.rb [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+end
+
+action :create do
+  current_resource = @current_resource
+  converge_by("Create #{@new_resource}") do
+    execute "systemctl daemon-reload" do
+      action :nothing
+    end
+
+    directory _get_override_directory do
```
Comment (AbelNavarro) [inline chef/cookbooks/utils/providers/systemd_environment.rb]: we may need to delete this directory on `:delete`

File: chef/cookbooks/utils/providers/systemd_environment.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+def _get_unit_name
+  "#{@new_resource.service_name}.service"
+end
+
+def _get_override_directory
+  "/etc/systemd/system/#{_get_unit_name}.d"
+end
+
+def _get_override_file_path
+  "#{_get_override_directory}/60-environment.conf"
```
Comment (AbelNavarro) [inline chef/cookbooks/utils/providers/systemd_environment.rb]: Is this file name the same for all services?

Comment (nicolasbock) [reply chef/cookbooks/utils/providers/systemd_environment.rb]: No this is specific to this override here.

File: chef/cookbooks/utils/providers/systemd_environment.rb (hunk 1 above)
Comment (nicolasbock) [reply chef/cookbooks/utils/providers/systemd_environment.rb]: Good point

**Comment by nicolasbock** [comment]
@AbelNavarro TBH I copy and pasted that from an existing `systemd` provider. I 'll check what the `converge_by` primitive does. Thanks!

**Comment by nicolasbock** [comment]
@AbelNavarro Here is a nice explanation: https://stackoverflow.com/questions/35238385/chef-what-is-meant-by-converge-by

**Comment by nicolasbock** [comment]
@AbelNavarro I added your suggested directory delete. Please have another look.

**Comment by nicolasbock** [comment]
Thanks @AbelNavarro . It's not urgently needed at the moment, but could be useful I suppose.

**Comment by JanZerebecki** [comment]
This branch is closed for new features. Please reopen if you feel otherwise.

**Pull request closed without merging**